pyam/core.py

```diff
@@ -261,34 +261,43 @@ def metadata(self, meta, name=None):
             (by `['model', 'scenario']` index if possible)
         name: str
             category column name (if not given by meta pd.Series.name)
+        index: pd.MultiIndex
+            index to be used for setting metadata column
         """
-        if isinstance(meta, pd.Series) and \
-                set(['model', 'scenario']).issubset(meta.index.names):
-            meta.name = name or meta.name
+        # create pd.Series from index and meta if provided
+        _meta = meta if index is None else pd.Series(data=meta, index=index)
+
+        # append by index of pd.Series if possible
+        if isinstance(_meta, pd.Series) and \
+                set(['model', 'scenario']).issubset(_meta.index.names):
+            _meta.name = name or _meta.name
             # reduce index dimensions to model-scenario only
-            meta = (meta
-                    .reset_index()
-                    .loc[:, ['model', 'scenario', meta.name]]
-                    .set_index(['model', 'scenario'])
-                    )
+            _meta = (_meta
```

Comment:
I think this can be `meta = meta.drop(index=[other indicies])[name]`

Reply:
that wouldn't work when passing over an index with more than 2 levels, need to bring it downb to the META_IDX first

Reply:
another consideration, the `s[name]` doesn't work on `pd.Series`, so the reindexing also takes care of a harmonization to a `pd.DataFrame`